### PR TITLE
[macOS - Feat] Add support for Apple Gaming porting toolkit

### PR DESCRIPTION
**Pull request by flavioislima**
With this, Hyperplay will look for a current setup of the apple's GPTK and list it under the Wine List

## HOW TO TEST: Downloaded Games

- To download the build to test, go bellow under 'CHECKS' and build_macOS click on `details`. Then on the page that will open, click on SUMMARY. And then go to ARTIFACTS and download the build for your system.
- Since this build is NOT signed (only releases are), you need to first run this command on the terminal: `xattr -d -r com.apple.quarantine PATH_TO/HyperPlay.app`
- Make sure you have installed the gaming porting toolkit following the Install instructions provided by Apple.
- Make sure Spotlight is enabled and the command `mdfind wine64` returns something on the terminal.
- Try to run some games and make sure it is working already.
- When opening HyperPlay, go to Settings and on Game default check if, under WINE, it shows the GPTK option on the list

- Download and install some Windows games and select the Script you want to use on the game wine settings.
- Use new prefixes instead of reusing an existing one. If you have DXVK installed on the prefix, the game might not work.

## Sideloaded Games

- Tested only Steam so far
- Click on ADD GAMES on the library.
- Fill in the title as you want.
- Select the Gaming Toolkit script you want to use.
- Click on RUN INSTALLER FIRST and select the exe you want to install
- The app will be installed inside the prefix
- On Select EXE, select the exe to run the app, for instance: `/Users/name/Games/GPT_PREFIX/drive_c/Program Files (x86)/Steam/Steam.exe`
- Click Finish

---

Use the following Checklist if you have changed something on the Backend or Frontend:

- [x] Tested the feature and it's working on a current and clean install.
- [ ] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

### CODE REVIEW

File: src/backend/launcher.ts [hunk 1]
```diff
@@ -313,7 +307,7 @@ function setupWineEnvVars(gameSettings: GameSettings, gameId = '0') {
   }
 
   if (gameSettings.showFps) {
-    ret.DXVK_HUD = 'fps'
+    isMac ? (ret.MTL_HUD_ENABLED = '1') : (ret.DXVK_HUD = 'fps')
```
Comment (eliobricenov) [inline src/backend/launcher.ts:310]: This is my preference but using ternaries for branching is awkward. I know this is for the assignment of different `ret` fields but I'd do:
```typescript
if (isMac) {
  ret.MTL_HUD_ENABLED = '1'
} else {
  ret.DXVK_HUD = 'fps'
}

```

**Review by eliobricenov — APPROVED** [review]
LGTM code-wise. Just one personal preference that you can omit

File: src/backend/launcher.ts (hunk 1 above)
Comment (flavioislima) [reply src/backend/launcher.ts]: done

**Review by BrettCleary — APPROVED** [review]
I can confirm that my GPT install is found successfully and the GPT commands run correctly. There are still many games that do not launch after this, but I think that is a GPT issue and not an issue with your branch

**Review by nyghtstalker — APPROVED** [review]
@flavioislima go ahead and give it your best once over since you have it working the best and understand it the most. After that, we can merge it. I would prefer to get you unblocked and continue working.

**Pull request merged**